go-controller/pkg/ovn/controller/egressservice/egressservice_zone.go

```diff
@@ -585,41 +724,91 @@ func (c *Controller) syncEgressService(key string) error {
 	}
 
-	// At this point the states are valid and we should create the proper logical router policies.
+	// At this point the states are valid and we should create the proper logical router policies and static routes.
 	// We reach the desired state by fetching all of the endpoints associated to the service and comparing
 	// to the known state:
 	// We need to create policies for endpoints that were fetched but not found in the cache,
 	// and delete the policies for those which are found in the cache but were not fetched.
 	// We do it in one transaction, if it succeeds we update the cache to reflect the new state.
 
-	v4ToAdd := v4Endpoints.Difference(state.v4Endpoints).UnsortedList()
-	v6ToAdd := v6Endpoints.Difference(state.v6Endpoints).UnsortedList()
-	v4ToRemove := state.v4Endpoints.Difference(v4Endpoints).UnsortedList()
-	v6ToRemove := state.v6Endpoints.Difference(v6Endpoints).UnsortedList()
+	v4LocalToAdd := v4LocalEndpoints.Difference(state.v4LocalEndpoints).UnsortedList()
+	v6LocalToAdd := v6LocalEndpoints.Difference(state.v6LocalEndpoints).UnsortedList()
+	v4LocalToRemove := state.v4LocalEndpoints.Difference(v4LocalEndpoints).UnsortedList()
+	v6LocalToRemove := state.v6LocalEndpoints.Difference(v6LocalEndpoints).UnsortedList()
+
+	v4RemoteToAdd := v4RemoteEndpoints.Difference(state.v4RemoteEndpoints).UnsortedList()
+	v6RemoteToAdd := v6RemoteEndpoints.Difference(state.v6RemoteEndpoints).UnsortedList()
+	v4RemoteToRemove := state.v4RemoteEndpoints.Difference(v4RemoteEndpoints).UnsortedList()
+	v6RemoteToRemove := state.v6RemoteEndpoints.Difference(v6RemoteEndpoints).UnsortedList()
+
+	// v[4|6]LocalEndpoints represents endpoints local to the current zone.
+	// v[4|6]RemoteEndpoints represents endpoints remote to the current zone.
+	// If a service is hosted in the local zone:
+	//  - create LRPs for local endpoints with mgmt IP as a nextHop
+	//  - create LRSRs for remote endpoints with mgmt IP as a nextHop
+	// If a service is hosted in a remote zone:
+	//  - create LRPs for local endpoints with transit switch IP as a nextHop
+	//  - do nothing for remote endpoints
+	// When IC is disabled v[4|6]RemoteEndpoints are empty,
+	// service is considered to be local and LRSRs are not modified.
+
+	nextHopV4 := node.v4MgmtIP.String()
+	nextHopV6 := node.v6MgmtIP.String()
+	svcNodeInLocalZone := true
+	if config.OVNKubernetesFeature.EnableInterconnect {
+		var zoneKnown bool
+		zoneKnown, svcNodeInLocalZone = c.nodeZoneState(node.name)
+		if !zoneKnown {
+			return fmt.Errorf("failed to verify whether the svc node %s is in the local zone", node.name)
+		}
+		if !svcNodeInLocalZone {
+			nextHopV4 = node.transitIPV4.String()
+			nextHopV6 = node.transitIPV6.String()
+		}
+	}
 
 	allOps := []libovsdb.Operation{}
-	createOps, err := c.createLogicalRouterPoliciesOps(key, node.v4MgmtIP.String(), node.v6MgmtIP.String(), v4ToAdd, v6ToAdd)
+	createOps, err := c.createLogicalRouterPoliciesOps(key, nextHopV4, nextHopV6, v4LocalToAdd, v6LocalToAdd)
```

Comment:
should be createOrUpdate? as it will also update the next hop if that has changed (like another node was egress service node, but now this one is). Maybe a comment explaining that would be good too.